Tools/scripts/uploader.py

```diff
@@ -794,14 +851,15 @@ def main():
     parser.add_argument('--source-system', type=int, action="store", help="Source system to send reboot mavlink packets from", default=255)
     parser.add_argument('--source-component', type=int, action="store", help="Source component to send reboot mavlink packets from", default=0)
     parser.add_argument('--download', action='store_true', default=False, help='download firmware from board')
-    parser.add_argument('firmware', action="store", help="Firmware file to be uploaded")
+    parser.add_argument('--identify', action="store_true", help="Do not flash firmware; simply dump information about board")
+    parser.add_argument('firmware', nargs="?", action="store", default=None, help="Firmware file to be uploaded")
```

Comment:
Given that firmware is now optional, shouldn't we add a new check that firmware is not None unless command is identify?

Reply:
Yes, I've added a check, thanks!